### DISCUSSION TITLE
403 forbidden

### DISCUSSION THREAD

**Discussion by NeedNot** [userbradley/BeReal#49] [Q&A] [answered]
anyone else getting this? not sure why but this morning the api just stopped working and this is all i get
```
<html>

<head>
    <title>403 Forbidden</title>
</head>

<body>
    <center>
        <h1>403 Forbidden</h1>
    </center>
    <hr>
    <center>nginx</center>
</body>

</html>
```

**Comment by NeedNot** [comment, marked as answer]
welp. it's been fun guys

**Reply by NicoWeio** [reply]
_As somebody who isn't familiar with this repository:_
This sounds like you are giving up. What is it that makes you think you can't just update to their (apparently) new API?

**Reply by NeedNot** [reply]
since october all requests sent a "bereal-signature" header with what appears to be a hash and now today it's being enforced. we don't know how they are generating the signature. If someone could reverse engineer the algorithm used than that's would be great but reading obfusicated code isn't very easy

**Reply by vexcited** [reply]
See https://github.com/userbradley/BeReal/discussions/24#discussioncomment-12366415

**Comment by bengram** [comment]
it was a nice time. it's over

**Reply by NeedNot** [reply]
yeah. idk what you guys did with it but i had some innocent fun. maybe i should reveal idk

**Reply by MrAn0nym** [reply]
> yeah. idk what you guys did with it but i had some innocent fun. maybe i should reveal idk

Had 2 extremely inefficient scripts that allowed me to inject in Java written code, similar to how aliucord worked. Every once in a while when BeReal updated I completely forgot how the scripts worked and had to spend 30mins to understand them again 😂

**Reply by U14-dev** [reply]
Uhh sounds cool tho^ Never injected directly into the apk but this might solve our problems since we can use the official signature?  What is your dc @ ?

**Comment by U14-dev** [comment]
Same here, had some fun while it lasted. Last working API call was about 6h ago^

**Comment by NeedNot** [comment]
i had hit the friend limit of 1000 so i stopped accepting them and let them accumulate. just added them all and the api says i have 1538 friends. and i never requested any of them

**Reply by U14-dev** [reply]
yooo 1000+ friends is crazy

**Reply by NeedNot** [reply]
when friends of friends became a thing it exploded. at one time 150 requests per day

**Reply by U14-dev** [reply]
hehe yeah, i had gif reactions under my friend of friends posts and everybody was asking me how :D
I really hope that we can get around this somehow, but the chances are not that great...

**Reply by NeedNot** [reply]
gif reactions, those were an actual feature which was in the android app a few years ago. i managed to achieve animated webps which were actually better since they have more animation customization and higher quality. I set the first frame to 1ms and the second frame to 5 minutes this made ios and android see a completely different image since ios doesn't support animated webps or even the gifs

**Reply by U14-dev** [reply]
nice one^ Do you have any of the animated stuff saved in your acc? I have one gif set as quick reaction so i can keep using it now xd

**Reply by NeedNot** [reply]
all of my quick reactions are yeah, how i never use them since i instant react to everyone, no one questions how i react to 1500 people per day with a unique image every time.

the good news is you should be able to edit a request to contain an already uploaded image so you could go through memories and find the reaction url and then edit a request to contain it. this is ofc you have a second account which was reacted to

**Reply by U14-dev** [reply]
But how can i get the url of the reaction within the app?

**Reply by NeedNot** [reply]
android emulator with root access, talked about on these other discussions. then use http toolkit. you can set a mock which pauses a specific url request and lets you modify the response or the request. request wont work because of the signature but modifying the response of the image upload url might

**Reply by U14-dev** [reply]
ah yeah sure just sounded like you can get urls within the official app which made me curios. But editing reqs will be a lot more work than playing with the official api so idk if i'll do that

**Comment by U14-dev** [comment]


**Reply by NeedNot** [reply]
yeah i found that out too, they aren't enforcing the signature integrity atm

**Reply by NeedNot** [reply]
bereal employees probably reading our msgs right now

**Reply by NeedNot** [reply]
says you aren't accepting, @need_not

**Reply by NicoWeio** [reply]
> bereal employees probably reading our msgs right now

With that in mind, do you think there is a way to make a fix available to me and others? Feel free to text me _@nico_weio_ if you deem that necessary.

**Reply by NicoWeio** [reply]
@NeedNot I saw your :+1: here, but I don't see anything on Discord. Just FYI.

**Reply by U14-dev** [reply]
Yeah we basically did the same as rvaidun but since the bereal team is watching i thought i might not share it here to prevent any further damages^

**Reply by NicoWeio** [reply]
Sorry for stepping on your toes then. It's just my stance that it's impossible to have fixes propagate in open-source projects without BeReal guys noticing.

**Reply by MrAn0nym** [reply]
Had a reason why I suggested creating a discord server instead of public GitHub discussions 😆😂

**Reply by NicoWeio** [reply]
From what the previous change looked like, I think it's clear that they would have implemented actual signing eventually. 
So is this where we're at? Is there still some motivation to create a Discord server?

**Reply by bengram** [reply]
if there is a discord server i will be happy to participate

**Reply by max21910** [reply]
As userbradley say : 
"This is in response to https://github.com/userbradley/BeReal/issues/16 where a user had requested we create a Discord.

By keeping everything on GitHub, it enables us to build a community where everyone has access to the information, not just those with discord accounts"

**Comment by NicoWeio** [comment]
Here's how another project implemented a (basic) fix: https://github.com/rvaidun/befake/commit/6ae70ca3ecd8b90c1563c5d3939caa8326ce1f10

**Comment by MrAn0nym** [comment]
I stopped playing this cat and mouse game when they implemented pairip and switched to apk bundles
